### ISSUE TITLE
Legend is cropped from left & right if too large to fit in container (change in beta 31)

### ISSUE THREAD

**Issue by dswitzer** [amcharts/amcharts4#139]
The layout of the legends has changed after beta 31. Prior to beta 31, if a legend was not wide enough to fit, it was left aligned and the right-most portion of the legend was cropped.

Now the legend is centered aligned, so the text both on the right & left get cropped. 

There doesn't seem to be a way to revert back to the old behavior. If I try to set the contentAlign to "left" it does have the same behavior if the legend it too wide to fit, but I want it to be centered if it does fit.

Here's what the layout used to look like in beta 30:
![legend_beta_30](https://user-images.githubusercontent.com/900497/42584887-4155412a-8502-11e8-9cae-b4d355f42e06.png)

Here's what it looks like in beta 31:
![legend_beta_31](https://user-images.githubusercontent.com/900497/42584912-4969203e-8502-11e8-93a5-00c8d135da18.png)

The legend looks much better in beta 30, where the beginning of the legend is shown.

You can see an example of the issue here:
https://codepen.io/team/amcharts/pen/KeOPya?editors=0010

**Comment by martynasma** [comment]
Indeed. There was bound to be some issues due to reworked layouting.

I wouldn't not call it a bug just yet. Give us a few days to discuss this, an how to best go about it.

**Comment by dswitzer** [comment]
While it would obviously be best to have no cropping occur, the way it worked before was better.

The best would be to right crop the labels so that the values always show (which is something I opened up a ticket about at one point).

**Comment by arm1n** [comment]
I don't know if it's related, but since I upgraded to `4.0.0-beta.31` my right-aligned axis scale is cut off from screen. It's easy reproducible by uncommenting code `range.label.align = "right";` from the example in the [official docs](https://www.amcharts.com/docs/v4/concepts/axes/axis-ranges/#Adding_guide_labels).

**Comment by martynasma** [comment]
@arm1n This fix takes care of your issue as well.

**Comment by martynasma** [comment]
For the record, I noticed that legend now resizes when toggling its items. Not sure if it was like that before.

Opened another issue for that: #151

**Comment by dswitzer** [comment]
@martynasma 

Thanks! This appears fixed.
